src/openforms/formio/registry.py

```diff
@@ -52,18 +51,18 @@ class BasePlugin(Generic[T], AbstractBasePlugin):
 
     is_enabled: bool = True
 
-    formatter: type[FormatterProtocol]
+    formatter: type[FormatterProtocol[T]]
     """
     Specify the callable to use for formatting.
 
     Formatter (class) implementation, used by
     :meth:`openforms.formio.registry.ComponentRegistry.format`.
     """
-    normalizer: None | NormalizerProtocol = None
+    normalizer: None | NormalizerProtocol[T] = None
```

Comment:
```suggestion
    normalizer: NormalizerProtocol[T] | None = None
```

Feels weird to have `None` as the first type of the union?